session/storage_memory_test.go

```diff
@@ -111,9 +113,78 @@ func TestStorage_Remove_Does_Not_Panic(t *testing.T) {
 	assert.Len(t, storage.sessions, 1)
 }
 
+func TestStorage_PublishesEventsOnCreate(t *testing.T) {
+	instance := expectedSession
+	mp := &mockPublisher{}
+	sessionStore := NewStorageMemory(mp)
+
+	sessionStore.Add(instance)
+
+	// since we're shooting the event in an asynchronous fashion, try every millisecond to see if we already have it
+	attempts := 0
+	// functions are really just easier to exit from than nested blocks
+	func() {
+		for {
```

Comment:
This can be a `for` with `range`: 
```go
	for range time.After(time.Microsecond) {
		if attempts > 1000 {
			assert.Fail(t, "no change after a 1000 attempts")
			break
		}
		attempts++
		if mp.getLast().Action == sessionEvent.Created {
			break
		}
	}
```